packages/libs/client-utils/src/nodejs/yaml-converter.ts

```diff
@@ -0,0 +1,213 @@
+import type {
+  ChainId,
+  IExecutionPayloadObject,
+  IPactCommand,
+} from '@kadena/client';
+import { readFileSync } from 'fs';
+import yaml from 'js-yaml';
+import { join } from 'path';
+import { asyncPipe } from '../core/utils/asyncPipe';
+
+interface ITplHoleTriple {
+  literal: string;
+}
+interface ITplHoleDouble {
+  parsed: string;
+}
+type TplHole = ITplHoleTriple | ITplHoleDouble;
+type TplPart = string;
+type PartsAndHoles = [TplPart[], TplHole[]];
+
+interface ITemplateContext {
+  cwd: string;
+  tplPath: string;
+  tplString: PartsAndHoles;
+}
+
+type FilledYamlString = string;
+interface ITemplateContextReplacedHoles extends ITemplateContext {
+  filledYamlString: FilledYamlString;
+}
+
+interface ITemplateTransaction {
+  codeFile?: string;
+  command: string;
+  meta: {
+    chainId: string;
+    sender: string;
+    gasLimit: number;
+    gasPrice: number;
+    ttl: number;
+  };
+  networkId: string;
+  data: Record<string, any>;
+  signers: Array<{ public: string }>;
+  nonce: string;
+}
+
+interface ITemplateContextPactCommand extends ITemplateContextReplacedHoles {
+  tplTx: Omit<ITemplateTransaction, 'codeFile'> & { code?: string };
+}
+
+// Responsinble for splitting a string into parts and holes
+export const getPartsAndHoles = (text: string) => {
+  return text.split(/{{(.*?)}}}?/g).reduce(
+    (acc, curr, i) => {
+      // if i is even, then it's a part
+      // if i is odd, then it's a hole
+      if (i % 2 === 0) {
+        acc[0].push(curr);
+      } else {
+        if (curr.startsWith('{')) {
+          acc[1].push({ literal: curr.slice(1) });
+          return acc;
+        }
+        acc[1].push({ literal: curr });
+      }
+      return acc;
+    },
+    [[], []] as PartsAndHoles,
+  );
+};
+
+export const getPartsAndHolesInCtx = (
+  tplPath: string,
+  cwd: string = process.cwd(),
+): ITemplateContext => {
+  const file = readFileSync(join(cwd, tplPath), 'utf-8').toString();
+  const tplString = getPartsAndHoles(file);
+
+  return {
+    tplPath,
+    cwd,
+    tplString,
+  };
+};
+
+// Responsible for replacing holes with values
+export const replaceHoles = (
+  partsAndHoles: PartsAndHoles,
+  args: Record<string, string | number>,
+) => {
+  const [parts, holes] = partsAndHoles;
+  const allParts = zip(parts, holes);
+  return allParts
+    .map((partOrHole, index) => {
+      if (typeof partOrHole === 'string') {
+        // it's a part
+        return partOrHole;
+      } else {
+        if ('literal' in partOrHole) {
+          // it's a literal hole
+          if (!(partOrHole.literal in args)) {
+            throw new Error(
+              `argument to fill hole for ${partOrHole.literal} is missing in ${
+                allParts[index - 1]
+              }{{${partOrHole.literal}}}${allParts[index + 1]}}`,
+            );
+          }
+
+          return args[partOrHole.literal];
+        }
```

Comment:
Maybe we can add a note that we don't know what the exact difference is in kda-tool for `{{}}` and `{{{}}}`, and that we currently handle everything as `literal`

Reply:
Good idea. Done 👍🏽